plugins/modules/mso_schema_template_migrate.py

```diff
@@ -0,0 +1,173 @@
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: mso_schema_template_migrate
+short_description: Migrate Bridge Domains (BDs) and EPGs between templates
+description:
+- Migrate BDs and EPGs between templates of same and different schemas.
+author:
+- Anvitha Jain (@anvitha-jain)
+options:
+  schema:
+    description:
+    - The name of the schema.
+    type: str
+    required: yes
+  template:
+    description:
+    - The name of the template.
+    type: str
+    required: yes
+  bds:
+    description:
+    - The name of the BDs to migrate.
+    type: list
+    elements: str
+  epgs:
+    description:
+    - The name of the EPGs to migrate.
+    type: list
+    elements: dict
+  target_schema:
+    description:
+    - The name of the target_schema.
+    type: str
+    required: yes
+  target_template:
+    description:
+    - The name of the target_template.
+    type: str
+    required: yes
+  state:
+    description:
+    - Use C(present) for adding .
+    type: str
+    default: present
+extends_documentation_fragment: cisco.mso.modules
+'''
+
+EXAMPLES = r'''
+- name: Migration of objects between templates of same schema
+  mso_schema_template_migrate:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    target_schema: Schema 1
+    target_template: Template 2
+    bds:
+    - BD
+    epgs:
+    - epg: EPG1
+    anp: ANP
+    state: present
+  delegate_to: localhost
+
+- name: Migration of objects between templates of different schema
+  mso_schema_template_migrate:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    target_schema: Schema 2
+    target_template: Template 2
+    bds:
+    - BD
+    epgs:
+    - epg: EPG1
+    anp: ANP
+    state: present
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec, mso_reference_spec, mso_subnet_spec, mso_dhcp_spec
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        schema=dict(type='str', required=True),
+        template=dict(type='str', required=True),
+        bds=dict(type='list', elements='str'),  # This parameter is not required for querying all objects
+        epgs=dict(type='list', elements='dict'),
+        target_schema=dict(type='str', required=True),
+        target_template=dict(type='str', required=True),
+        state=dict(type='str', default='present'),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['bds', 'epgs']],
+            ['state', 'present', ['bds', 'epgs']],
```

Comment:
for 'present' state, are 'bds' and 'epgs' both required?
if both of them are required, we can just add 'required=True' in argument_spec.update() since we only have 'present' state.
if they are not required, just remove this line

Reply:
Done